### PR TITLE
tank now sends mqtt status message

### PR DESCRIPTION
**Pull request by immunity-anomaly**
Tank sends status message when state changes.

### CODE REVIEW

File: vosekast_control/Scale.py [hunk 1]
```diff
@@ -260,30 +263,34 @@ def add_new_value(self, new_value):
...
         # volume_flow = calculated volume flow
-        try:
-            mqttmsg = StatusMessage(
-                "scale", new_value, "Kg", volume_flow, "L/s")
-        except:
-            mqttmsg = StatusMessage("scale", new_value, "Kg", None, None)
 
-        if self.mqtt.connection_test():
-            self.mqtt.publish_message(mqttmsg)
+        if self.scale_publish != True:
```
Comment (karlTGA) [inline vosekast_control/Scale.py]: You can write that shorter:
``` python 
if not self.scale_publish:
```

File: vosekast_control/Tank.py [hunk 2]
```diff
@@ -187,3 +187,8 @@ def state(self):
     def state(self, new_state):
         self._state = new_state
         self.logger.info(f"New Tank state is: {new_state}")
+
+        mqttmsg = StatusMessage(
+                    self.name, f"New Tank state is: {new_state}", None, None, None)
```
Comment (karlTGA) [inline vosekast_control/Tank.py]: Has a status message not a clear value? That looks for me like a log message.

File: vosekast_control/Tank.py [hunk 3]
```diff
@@ -187,3 +187,8 @@ def state(self):
...
+        if self.mqtt.connection_test():
```
Comment (karlTGA) [inline vosekast_control/Tank.py]: Instat that you call every time the connection test before you send a message, you can make that test in the publish function. That is much clearer.

File: vosekast_control/Scale.py [hunk 4]
```diff
@@ -44,6 +44,7 @@ def __init__(
         self.vosekast = vosekast
         self.state = States.NONE
         self.mqtt = self.vosekast.mqtt_client
+        self.scale_publish = False
```
Comment (karlTGA) [inline vosekast_control/Scale.py]: Whats the meaning of that new variable?

Comment (immunity-anomaly) [reply vosekast_control/Scale.py]: scale_publish allows the scale to only publish values if the input is valuable.

File: vosekast_control/Scale.py (hunk 1 above)
Comment (immunity-anomaly) [reply vosekast_control/Scale.py]: will be implemented in next commit

File: vosekast_control/Tank.py (hunk 2 above)
Comment (immunity-anomaly) [reply vosekast_control/Tank.py]: changed to send log message only

File: vosekast_control/Tank.py (hunk 3 above)
Comment (immunity-anomaly) [reply vosekast_control/Tank.py]: moved connection_test to publish function

File: vosekast_control/MQTT.py [hunk 5]
```diff
@@ -37,10 +37,13 @@ def __init__(self, host):
         try:
             await self.client.connect(self.host)
         except ConnectionRefusedError:
+
             await self.connection_refused()
                 
     async def connection_refused(self):
-        self.logger.warning("Connection refused. Is the MQTT broker installed? Retrying.")
+        self.logger.warning(
+          "Connection refused. Is the MQTT broker installed? Retrying.")
```
Comment (karlTGA) [inline vosekast_control/MQTT.py]: Maybe better `Is the MQTT Broker accessible?`

File: vosekast_control/MQTT.py [hunk 6]
```diff
@@ -54,11 +57,13 @@ def connected(self):
...
     def publish(self, topic, message):
-        self.client.publish(topic, message, qos=0)
-        # print("Published: \"" + message + "\" to topic: \"" + topic + "\"")
+        if self.connection_test():
+            self.client.publish(topic, message, qos=0)
+            # print("Published: \"" + message + "\" to topic: \"" + topic + "\"")
 
     def publish_message(self, message_object):
-        self.publish(message_object.topic, message_object.get_json())
+        if self.connection_test():
```
Comment (karlTGA) [inline vosekast_control/MQTT.py]: Is it necessary to check the connection twice?

File: vosekast_control/MQTT.py (hunk 5 above)
Comment (immunity-anomaly) [reply vosekast_control/MQTT.py]: ok

File: vosekast_control/MQTT.py (hunk 6 above)
Comment (immunity-anomaly) [reply vosekast_control/MQTT.py]: will be fixed soon, thank you.

**Pull request merged**